src/main/java/org/folio/dew/controller/MarcExportController.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.dew.controller;
+
+import static java.lang.String.format;
+import static org.folio.dew.utils.Constants.FILE_UPLOAD_ERROR;
+import static org.folio.dew.utils.Constants.PATH_SEPARATOR;
+import static org.folio.dew.utils.Constants.TMP_DIR_PROPERTY;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.io.FileUtils;
+import org.folio.dew.repository.MinIOObjectStorageRepository;
+import org.openapitools.api.UploadApi;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.annotation.PostConstruct;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+@RestController
+@RequestMapping("/marc-export")
+@Log4j2
+@RequiredArgsConstructor
+public class MarcExportController implements UploadApi {
+  private final MinIOObjectStorageRepository repository;
+
+  private String workDir;
+
+  @Value("${spring.application.name}")
+  private String springApplicationName;
+
+  @PostConstruct
+  public void postConstruct() {
+    workDir = System.getProperty(TMP_DIR_PROPERTY) + PATH_SEPARATOR + springApplicationName + PATH_SEPARATOR;
+  }
+
+  @Override
+  public ResponseEntity<String> uploadFile(MultipartFile file) {
+    if (file.isEmpty()) {
+      return new ResponseEntity<>(format(FILE_UPLOAD_ERROR, "file is empty"), HttpStatus.BAD_REQUEST);
+    }
+
+    var uploadedPath = Path.of(workDir, file.getOriginalFilename());
+
+    try {
+      if (Files.exists(uploadedPath)) {
+        FileUtils.forceDelete(uploadedPath.toFile());
+      }
+      Files.write(uploadedPath, file.getBytes());
```

Comment:
Why local files are used??

Reply:
Local files are used because the job works in scope of single request-response.

Reply:
@TarasSpashchenko Can expect IO operations performance decreasing due to using remote files storage because we are going to process uploaded file?